src/ape/api/providers.py

```diff
@@ -850,6 +852,37 @@ def send_call(self, txn: TransactionAPI, **kwargs) -> bytes:
             str: The result of the transaction call.
         """
 
+        if (
+            not isinstance(self, TestProviderAPI)
+            or not self.debug_trace_handlers
+            or not txn.receiver
+        ):
+            return self._send_call(txn, **kwargs)
+
+        # Handle call tree, if trace handlers set.
+        # For example, this is used for capturing gas usage in view calls
+        # during tests when the `--gas` flag was set.
+
+        try:
+            with self.chain_manager.isolate():
+                account = self.account_manager.test_accounts[0]
+                receipt = account.call(txn)
+                call_tree = receipt.call_tree
+                if not call_tree:
+                    return self._send_call(txn, **kwargs)
+
+                else:
+                    for handler in self.debug_trace_handlers.values():
+                        handler(
+                            call_tree, txn.sender, txn.receiver, transaction_hash=receipt.txn_hash
+                        )
+
+                    return call_tree.returndata
+
+        except APINotImplementedError:
+            return self._send_call(txn, **kwargs)
```

Comment:
What can raise `APINotImplementedError` in the block above? We might wanna wrap the try/except around only the parts that can raise if possible

Reply:
it is the `__enter__` method of the `with` statement, so I think it has include everything in the with part too

Reply:
Also could happen in the `__exit__` maybe

Reply:
I made the stuff in the middle into a helper method so it is more obvious that the try /except is about the `with isolate` line.